Project-Management/TechnicalSteeringCommittee/reports/Linode_Upgrade_Precise_To_Xenial.md

```diff
@@ -0,0 +1,171 @@
+Summary
+-------
+
+This document describes the process that will take place to migrate the Linode server from Ubuntu Precise Pangolin to Ubuntu Xenial Xerus.
+
+Motivation
+----------
+
+The Linode server is currently running Ubuntu Precise Pangolin (12.04). This will reach EOL (end-of-life) in April 2017 and so will no longer
+receive any further updates, security or otherwise. The server will be migrated to the latest LTS version of Ubuntu whose EOL date is April 2021.
+
+Description
+-----------
+
+The Linode machine currently provides the following services:
+
+* Wiki - www.mantidproject.org
+* Static docs - docs.mantidproject.org
+* Doxygen - doxygen.mantidproject.org
+* Jenkins - builds.mantidproject.org
+* Leeroy - builds.mantidproject.org:5000
+* Forums - forums.mantidproject.org/help.mantidproject.org
+* Download page - download.mantidproject.org
+* Script repository downloading - download.mantidproject.org/scriptrepository
+* Script repository upload - upload.mantidproject.org
+* Instrument schema - schema.mantidproject.org
+* External Data Central Repository - 198.74.56.37/ftp/external-data
+
+Transition
+==========
+
+There are 2 approaches to take when upgrading the OS:
+
+1. use the in-built `dist-upgrade` command to go `12.04`->`14.04`->`16.04`
+2. stand up a new linode instance, perform the setup, swap IP addresses with the original and delete the old one.
+
+The first approach requires 2 upgrade steps, introducing more risk into the process and might require just as much time if the upgrade goes
+wrong and a clean setup is required anyway.
+
+The second approach will allow more flexibility to keep existing services running and only have minimal downtime. Linode would charge `$80` at first
+for a new instance but would refund any time until the end of the next billing cycle if it is deleted. This seems like the most sensible option
+as it is very low cost and provides fallback.
+
+### Common system packages required:
+
+* `git`
+* Java
+* Docker
+
+### Current ppa list:
+
+Apt Configuration
+-----------------
+
+### Additions to `/etc/sources.list`
+
+```
+# Jenkins LTS repo
+deb http://pkg.jenkins-ci.org/debian-stable binary/
+```
+
+### Additional files in `/etc/sources.list.d`:
+
+### `docker.list`
+
+```
+deb https://get.docker.com/ubuntu docker main
+```
+
+### `git-core-ppa-precise.list`
+
+```
+deb http://ppa.launchpad.net/git-core/ppa/ubuntu precise main
+deb-src http://ppa.launchpad.net/git-core/ppa/ubuntu precise main
+```
+
+### `longview.list`
+
+```
+deb http://apt-longview.linode.com/ precise main
+```
+
+### `webupd8team-java-precise.list`
```

Comment:
True. https://support.cloudbees.com/hc/en-us/articles/203601234-CloudBees-Jenkins-Platform-Supported-Java-Versions says that both Oracle and OpenJDK are supported and Xenial has OpenJDK 8 so we shouldn't need this anymore.